cypress/e2e/tests/migration/migration-waves/export_to_jira_datacenter.test.ts

```diff
@@ -0,0 +1,131 @@
+        login();
+
+        jiraCredentials = new JiraCredentials(
+            data.getJiraCredentialData(CredentialType.jiraToken, true)
+        );
+        jiraCredentials.create();
+
+        jiraInstance = new Jira(
+            data.getJiraConnectionData(jiraCredentials, JiraType.server, false, true)
+        );
+        jiraInstance.create();
+    });
+
+    Object.values(JiraIssueTypes).forEach((issueType) => {
+        it(`Create wave to export as ${issueType}`, function () {
+            const apps = createMultipleApplications(2);
+            applications.push(...apps);
+
+            const migrationWave = new MigrationWave(
+                data.getRandomWord(8),
+                now,
+                end,
+                null,
+                null,
+                apps
+            );
+            migrationWave.create();
+            wavesMap[issueType] = migrationWave;
+        });
+    });
+
+    Object.values(JiraIssueTypes).forEach((issueType) => {
+        it(`Export wave as ${issueType} to Jira`, function () {
+            jiraInstance
+                .getProjectById(Cypress.env("jira_stage_datacenter_project_id"))
+                .then((project) => {
+                    expect(!!project).to.eq(true);
+
+                    projectName = project.name;
+
+                    return jiraInstance.getIssueType(issueType);
+                })
+                .then((issue) => {
+                    expect(!!issue).to.eq(true);
+
+                    wavesMap[issueType].exportToIssueManager(
+                        JiraType.server,
+                        jiraInstance.name,
+                        projectName,
+                        issue.name
+                    );
+                });
+        });
+    });
+
+    Object.values(JiraIssueTypes).forEach((issueType) => {
+        it(`Assert exports for ${issueType}`, function () {
+            cy.wait(40 * SEC); // Enough time to create both tasks and for them to be available in the Jira API
+            jiraInstance.getIssues(projectName).then((issues: JiraIssue[]) => {
+                const waveIssues = issues.filter((issue) => {
+                    return (
+                        (issue.fields.summary.includes(wavesMap[issueType].applications[0].name) ||
+                            issue.fields.summary.includes(
+                                wavesMap[issueType].applications[1].name
+                            )) &&
+                        issue.fields.issuetype.name.toUpperCase() ===
+                            (issueType as string).toUpperCase()
+                    );
+                });
+
+                jiraInstance.archiveIssues(waveIssues.map((issue) => issue.id));
+
+                expect(waveIssues).to.have.length(2);
+            });
+        });
+    });
+
+    after("Clear test data", function () {
+        MigrationWave.open(true);
+        Object.values(wavesMap).forEach((wave: MigrationWave) => wave.delete());
```

Comment:
Can we use deleteByList(waves) and deleteByList(applications) ?

Reply:
Sure, I'll change it